crypto/account.go

```diff
@@ -50,6 +51,35 @@ func GenerateAccount() (kp Account) {
 	return
 }
 
+// AccountFromPrivateKey derives the remaining Account fields from only a private key
+func AccountFromPrivateKey(sk ed25519.PrivateKey) (account Account, err error) {
+	if len(sk) == ed25519.SeedSize {
```

Comment:
How does this happen? It looks like `NewKeyFromSeed` accepts a byte array, so if we've already converted the seed into `ed25519.PrivateKey` doesn't that mean it's already gone through this conversion?

Reply:
Good point, I misinterpreted some things about `ed25519.PrivateKey` when I wrote this. I will update this function to just return an error if the length of `sk` is not `ed25519.PrivateKeySize`.